src/components/orchestrator/Orchestrator.test.tsx

```diff
@@ -311,4 +353,139 @@ describe('Orchestrator', () => {
+    const updateDataAndStateData = vi
+      .fn()
+      .mockRejectedValue(new Error('Network error'))
+
+    const { getByText } = renderWithRouter(
+      <QueryClientProvider client={queryClient}>
+        <Orchestrator
+          metadata={metadata}
+          mode={MODE_TYPE.COLLECT}
+          isDownloadEnabled={false}
+          initialInterrogation={interrogation}
+          // @ts-expect-error: we should have a better lunatic mock
+          source={sourceMultipleQuestion}
+          getReferentiel={() => {
+            return new Promise(() => [])
+          }}
+          updateDataAndStateData={updateDataAndStateData}
+          getDepositProof={() => {
+            return new Promise<void>(() => {})
+          }}
+        />
+      </QueryClientProvider>,
+    )
+
+    act(() => getByText('Start').click())
+    act(() => getByText('Continue').click())
+
+    const questionElement = getByText('my-question')
+    await user.click(questionElement)
+    await user.keyboard('test-input')
+
+    expect(document.title).toContain('*')
+
+    act(() => getByText('Continue').click())
+
+    await waitFor(() => {
+      expect(updateDataAndStateData).toHaveBeenCalledTimes(1)
+    })
+
+    expect(updateDataAndStateData).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        data: expect.objectContaining({
+          'my-question-input': {
+            COLLECTED: 'test-input',
+          },
+        }),
+      }),
+    )
+
+    //Wait for the request to fail
+    await waitFor(
+      () => {
+        expect(document.title).not.toBe('Sending the questionnaire')
+        expect(document.title).toContain('*')
+      },
+      { timeout: 3000 },
+    )
+  })
+
+  it('retains user input after failed save and successful retry', async () => {
+    const user = userEvent.setup()
+    let callCount = 0
+
+    // First call fails, second call succeeds
+    const updateDataAndStateData = vi.fn().mockImplementation(() => {
+      callCount++
+      if (callCount === 1) {
+        return Promise.reject(new Error('Network error'))
+      }
+      return Promise.resolve()
+    })
+
+    const { getByText } = renderWithRouter(
+      <QueryClientProvider client={queryClient}>
+        <Orchestrator
+          metadata={metadata}
+          mode={MODE_TYPE.COLLECT}
+          isDownloadEnabled={false}
+          initialInterrogation={interrogation}
+          // @ts-expect-error: we should have a better lunatic mock
+          source={sourceMultipleQuestion}
+          getReferentiel={() => {
+            return new Promise(() => [])
+          }}
+          updateDataAndStateData={updateDataAndStateData}
+          getDepositProof={() => {
+            return new Promise<void>(() => {})
+          }}
+        />
+      </QueryClientProvider>,
+    )
+
+    act(() => getByText('Start').click())
+    act(() => getByText('Continue').click())
+
+    const questionElement = getByText('my-question')
+    await user.click(questionElement)
```

Comment:
```suggestion
    await user.click(getByText('my-question'))
```

You can simplify this.